docs: Docs rework

agardnerIT opened the pull request:
## This PR
Attempts to re-structure the readmes to:

- Create a funnel whereby the main readme gets people started in the fastest way possible. So that:
- "Get me started" > "show me detailed documentation" > "show me really low level technical details"
- The main readme then links out to the existing detailed docs pages.

Aside from the main readme, the content hasn't changed much, but the intention is for it to be a bit more modular and easier to find things.

beeme1mr commented on docs/images/flagd-logical-architecture.jpg:
The "HTTP(S) or gRPC" part is not correct. That's how you access the API. The arrow represents a flag configuration syncs.

beeme1mr replied:
Please change "Kubernetes CRD" to "Kubernetes CR".

agardnerIT replied:
Is this now accurate?
<img width="1128" alt="image" src="https://user-images.githubusercontent.com/26523841/226762245-7b57b0f1-0cf1-49b7-9296-1bc799d996b1.png">

agardnerIT replied:
Committed new diagram after discussion this morning.

beeme1mr commented on docs/usage/evaluation_examples.md:
```diff
@@ -1,33 +1,6 @@
 # Evaluation Examples Using curl
 
-## Setup
```
Why was this section removed?

agardnerIT replied:
All commands are now available in the main readme and it makes this section redundant.

beeme1mr replied:
I think we'll need it when we incorporate these docs into the main doc page because it likely won't include the main readme. It's a bit redundant but I think that's okay in this case.

agardnerIT replied:
Added this back in. Resolving conversation.

beeme1mr commented on docs/configuration/fractional_evaluation.md:
```diff
@@ -1,5 +1,17 @@
 
+OpenFeature allows clients to pass contextual information which can then be used during a flag evaluation. For example, a client could pass the email address of the user.
+
+In some scenarios it is desirable to be use that contextual information to further segment the user population and thus return dynamic values.
```
```suggestion
In some scenarios, it is desirable to use that contextual information to segment the user population further and thus return dynamic values.
```

beeme1mr commented on docs/configuration/fractional_evaluation.md:
```diff
@@ -1,5 +1,17 @@
+
+In some scenarios it is desirable to be use that contextual information to further segment the user population and thus return dynamic values.
+
+Look at the [headerColor](https://github.com/open-feature/flagd/blob/main/samples/example_flags.flagd.json#L88-#L133) flag. The `defaultVariant` is `red` but it contains a [targeting rule](reusable_targeting_rules.md) meaning a fractional evaluation occurs for flag evaluation with a `context` object containing `email` and where that `email` value contains `@faas.com`.
```
```suggestion
Look at the [headerColor](https://github.com/open-feature/flagd/blob/main/samples/example_flags.flagd.json#L88-#L133) flag. The `defaultVariant` is `red`, but it contains a [targeting rule](reusable_targeting_rules.md), meaning a fractional evaluation occurs for flag evaluation with a `context` object containing `email` and where that `email` value contains `@faas.com`.
```

beeme1mr commented on docs/configuration/fractional_evaluation.md:
```diff
@@ -1,5 +1,17 @@
+
+Look at the [headerColor](https://github.com/open-feature/flagd/blob/main/samples/example_flags.flagd.json#L88-#L133) flag. The `defaultVariant` is `red` but it contains a [targeting rule](reusable_targeting_rules.md) meaning a fractional evaluation occurs for flag evaluation with a `context` object containing `email` and where that `email` value contains `@faas.com`.
+
+In this case, `25%` of the email addresses will receive `red`, `25%` will receive `blue` and so on.
```
```suggestion
In this case, `25%` of the email addresses will receive `red`, `25%` will receive `blue`, and so on.
```

beeme1mr commented on README.md:
````diff
@@ -14,23 +14,73 @@
+2. Start flagd:
+
+    ```sh
+    // Start flagd
````
Please make these separate commands so a user can use the copy to clipboard button in GitHub.

beeme1mr commented on README.md:
````diff
@@ -14,23 +14,73 @@
+    {"value":"val1","reason":"DEFAULT","variant":"key1"}
+    ```
+
+    Updates to the underlying flag store (e.g. JSON file) are reflected by flagd in realtime. No restarts required.
````
I wonder if step one should be to download the config file locally. That would allow the user to modify the flag config.

beeme1mr reviewed (commented):
This is a great start! Here are some quick thoughts.

I'll try and circle back later today with some additional suggestions. Thanks for putting this together. 🙇

beeme1mr commented on README.md:
```diff
@@ -14,23 +14,109 @@
+
 - OpenFeature compliant with providers available in many languages
-- Multiple flag configuration sources including `files`, `http`, and `Kubernetes`
+- Multiple flag configuration sources, including `files`, `http`, `gRPC` and `Kubernetes` can be used simultaneously
```
```suggestion
- Multiple flag configuration sources, including `files`, `http`, `gRPC` and `Kubernetes`
```

I don't think you need that section because it's listed below.

beeme1mr commented on docs/images/flagd-logical-architecture.jpg:
Can you have the flag sources arrows come from the flag state?

beeme1mr commented on README.md:
```diff
@@ -14,30 +14,120 @@
 
 Our community meetings are held regularly and open to everyone.
-Check the [OpenFeature community calendar](https://calendar.google.com/calendar/u/0?cid=MHVhN2kxaGl2NWRoMThiMjd0b2FoNjM2NDRAZ3JvdXAuY2FsZW5kYXIuZ29vZ2xlLmNvbQ) for specific dates and for the Zoom meeting links.
+Check the [OpenFeature community page](https://docs.openfeature.dev/community/) for all the ways to get involved.
```
Oh, nice change 👍

beeme1mr commented on README.md:
```diff
@@ -14,30 +14,120 @@
-- Flexible targeting rules based on [JSON Logic](https://jsonlogic.com/)
-- Lightweight daemon, with an emphasis on performance
-- Native support for metrics using Prometheus
+Flagd is a simple command line tool for fetching and evaluating feature flags.
```
```suggestion
Flagd is a feature flag daemon with a Unix philosophy.
```

beeme1mr commented on README.md:
```diff
@@ -14,30 +14,120 @@
+- 🔄 Centralized flag management. Use multiple flag sources and vendors. [Flag values are automatically merged](docs/configuration/flag_configuration_merging.md).
+- 🕒 Feature Flag updates occur in near-realtime (HTTP) or realtime (gRPC)
+- 💪 Contains a [powerful and flexible rule targeting engine](docs/configuration/reusable_targeting_rules.md) and [deterministic percentage-based rollouts](https://github.com/open-feature/flagd/blob/main/docs/configuration/fractional_evaluation.md).
+- 🔦 [Flag statistics and metrics are exposed](https://example.com) and compatible with Prometheus.
```
Update the example link.

agardnerIT replied:
I'll remove it for now as we don't have the documentation ready. I'll create another PR when that doc is ready.

beeme1mr commented on README.md:
```diff
@@ -14,30 +14,120 @@
+
+Think of it as a ready-made, open source, OpenFeature compliant feature flag backend system.
+
+flagd:
+
```
```suggestion
```

beeme1mr commented on README.md:
```diff
@@ -14,30 +14,120 @@
+
+- 🌐 OpenFeature compliant and [speaks your language](docs/usage/flagd_providers.md).
+- 🆕 Easy to [extend to new languages](docs/other_resources/creating_providers.md).
+- 🔄 Centralized flag management. Use multiple flag sources and vendors. [Flag values are automatically merged](docs/configuration/flag_configuration_merging.md).
```
```suggestion
- 🔄 Aggregate multiple data sources.
```

Talkd to Todd about this. We preferred to keep it shorter and omit vendors for now.

beeme1mr commented on README.md:
```diff
@@ -14,30 +14,120 @@
+- 🌐 OpenFeature compliant and [speaks your language](docs/usage/flagd_providers.md).
+- 🆕 Easy to [extend to new languages](docs/other_resources/creating_providers.md).
+- 🔄 Centralized flag management. Use multiple flag sources and vendors. [Flag values are automatically merged](docs/configuration/flag_configuration_merging.md).
+- 🕒 Feature Flag updates occur in near-realtime (HTTP) or realtime (gRPC)
```
```suggestion
- 🕒 Feature Flag updates occur in near-realtime
```

beeme1mr commented on .github/workflows/release-please.yaml:
```diff
@@ -118,7 +118,7 @@ jobs:
             DATE=${{ steps.date.outputs.date }}
       - name: Install Cosign
-        uses: sigstore/cosign-installer@ca922a571676e06d3299cce5d520e178ba89a276
+        uses: sigstore/cosign-installer@00bf1366a3f8c043c1f6f802441642bced142f5c
```
Did you mean to bump this version?

beeme1mr commented on docs/images/flagd-logical-architecture.jpg:
Could you add gRPC as another flag source?

beeme1mr reviewed (commented):
I made a few changes to the main readme. Most notably, I changed the Killercode link to a badge. Are you okay with these changes? If so, please adjust the architecture diagram based on the feedback. I think we'll be good to go after that!

agardnerIT commented:
All changes are good.

1) Added the gRPC flag source
2) Added one sentence pointing to the killercoda tutorial (in addition to the badge).

I like the badge (nice touch!) but to me, a badge is a notification element and not an interactive element so I wouldn't think to click on it. IMO the text makes it explicit so I'd prefer to keep some form of textual link in the copy.

beeme1mr commented on docs/README.md:
```diff
@@ -2,29 +2,49 @@
-- [Evaluation examples](./usage/evaluation_examples.md)
+## Installation Options
+
+[See all flagd installation options](../docs/usage/installation_options.md)
```
```suggestion
[See all flagd installation options](usage/installation_options.md)
```

beeme1mr commented on docs/README.md:
```diff
@@ -2,29 +2,49 @@
+
+## Copy and Paste Evaluation Options
+
+[This page](../docs/usage/evaluation_examples.md) provides copy and paste evaluation examples.
```
```suggestion
[This page](usage/evaluation_examples.md) provides copy and paste evaluation examples.
```

beeme1mr commented on docs/README.md:
```diff
@@ -2,29 +2,49 @@
+
+## Application Integration
+
+Once flagd is running, your next step is to integrate it into you application. [This page](../docs/usage/flagd_providers.md) shows all available integration options (called providers) in a variety of languages.
```
```suggestion
Once flagd is running, your next step is to integrate it into you application. [This page](usage/flagd_providers.md) shows all available integration options (called providers) in a variety of languages.
```
